bin/cylc-unregister

```diff
@@ -56,55 +58,43 @@ def main():
 
     db = RegistrationDB(options.db)
 
-    dirs = db.unregister(args[0])
-
-    n = len(dirs)
-    if n == 0:
-        print 'No suites unregistered.'
-        sys.exit(0)
-
-    print len(dirs), 'suites unregistered.'
-
-    if options.obliterate and len(dirs) > 0:
-        for dir in dirs:
-            print 'DELETE ', dir
-
-        really_obliterate = False
-        if options.force:
-            really_obliterate = True
-        else:
-            if len(dirs) == 1:
-                words = "THIS SUITE DEFINITION"
+    unregistered_list, skipped_list = db.unregister(args[0])
+
+    print '%d suite(s) unregistered.' % len(unregistered_list)
+    if not unregistered_list or not options.obliterate:
+        sys.exit(bool(skipped_list))
+
+    for _, path in unregistered_list:
+        if not options.force:
+            response = raw_input('REALLY DELETE "%s"? (y/n/a) ' % (path))
+            if response == 'a':
+                options.force = True
+            elif response != 'y':
+                continue
+        print 'DELETE "%s"' % (path)
+        try:
+            rmtree(path)
+        except OSError as exc:
+            print >> sys.stderr, (
+                "ERROR, could not remove directory: " + path)
+            print >> sys.stderr, str(exc)
+            continue
+        # recursively remove empty superdirs
+        tmp = path
+        while True:
+            tmp = os.path.split(tmp)[0]
+            try:
+                os.rmdir(tmp)
+            except OSError:
+                break
             else:
-                words = "THESE SUITE DEFINITIONS"
-            response = raw_input(
-                "DO YOU REALLY WANT TO DELETE " + words + "? (y/n) ")
-            if response == 'y':
-                really_obliterate = True
-        if really_obliterate and len(dirs) > 0:
-            for dir in dirs:
-                try:
-                    rmtree(dir)
-                except OSError as exc:
-                    print >> sys.stderr, (
-                        "ERROR, could not remove directory: " + dir)
-                    print >> sys.stderr, str(exc)
-                    continue
-                # recursively remove empty superdirs
-                tmp = dir
-                while True:
-                    tmp = os.path.split(tmp)[0]
-                    try:
-                        os.rmdir(tmp)
-                    except OSError:
-                        break
-                    else:
-                        print 'Removed empty directory:', tmp
+                print 'Removed empty directory:', tmp
+    sys.exit(bool(skipped_list))
 
 if __name__ == "__main__":
     try:
         main()
     except Exception as exc:
         if cylc.flags.debug:
-            raise
+            traceback.print_exc()
```

Comment:
I was doing this here to allow me to print out a better trace back, and forgot to change it back. It should probably be done for all commands in a separate change.